front/app/containers/Admin/projects/project/files/components/FilesUpload/components/SelectedFile.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import {
+  Box,
+  Icon,
+  Text,
+  Spinner,
+  colors,
+  Select,
+  Tooltip,
+} from '@citizenlab/cl2-component-library';
+import { CLError } from 'typings';
+
+import useAddFile from 'api/files/useAddFile';
+
+import Error from 'components/UI/Error';
+
+import { useIntl } from 'utils/cl-intl';
+import { getBase64FromFile } from 'utils/fileUtils';
+
+import { FileWithMeta } from '../types';
+
+import messages from './messages';
+
+type Props = {
+  fileMeta: FileWithMeta;
+  projectId: string;
+  onStatusUpdate: (status: Partial<FileWithMeta>) => void;
+};
+
+const SelectedFile = ({ fileMeta, projectId, onStatusUpdate }: Props) => {
+  const { formatMessage } = useIntl();
+  const { mutateAsync } = useAddFile();
+  const { file, status, semanticType } = fileMeta;
+  const [apiErrors, setApiErrors] = useState<CLError[] | null>(null);
+
+  // Ref to track if the upload has started
+  const hasStarted = useRef(false);
+
+  // Function to upload the file
+  const uploadFile = useCallback(async () => {
+    const base64 = await getBase64FromFile(file);
+    await mutateAsync(
+      {
+        content: base64,
+        project: projectId,
+        name: file.name,
+      },
+      {
+        onError: (errors) => {
+          setApiErrors(errors.error);
+          onStatusUpdate({ status: 'error' });
+        },
+        onSuccess: () => {
+          onStatusUpdate({ status: 'uploaded' });
+        },
+      }
+    );
+  }, [file, projectId, mutateAsync, onStatusUpdate]);
+
+  // Effect to handle the file upload when the status changes to 'uploading'
+  useEffect(() => {
```

Comment:
I'm not sure I fully understand why this effect is needed. Could you explain?

Reply:
This is a side-effect when "Status" changes for the file (E.g. When file status changes from queued to uploading, we want to trigger the "uploadFile").

I think the useEffect is needed as we don't want to have this happen on render right (which might cause some unexpected behaviour)?

Maybe I'm missing something though :thinking: